File: src/main/kotlin/fr/jhelp/kotlinLightSamples/Car.kt
package fr.jhelp.kotlinLightSamples

class Car : Vehicule, Comparable<Car>
{
    val model: String

    constructor(model: String, year: Int) : super(year)
    {
        this.model = model
    }

    constructor(model: String) : super()
    {
        this.model = model
    }

    override fun equals(other: Any?): Boolean
    {
        if (null == other || other !is Car)
        {
            return false
        }

        val otherCar = other!! as Car
        return this.model == otherCar.model && this.year == otherCar.year
    }

    override fun compareTo(other: Car): Int
    {
        val comparison = this.year - other.year

        if (comparison != 0)
        {
            return comparison
        }

        return this.model.compareTo(other.model)
    }

    override fun toString(): String
    {
        return "My super car : ${this.model} on ${this.year}"
    }
}

File: src/main/swift/Tests/swiftTests/src/jhelp/kotlinLightSamples/Car.swift

public class Car : Vehicule, Comparable
, Equatable , CustomStringConvertible {
     public var description: String { return self.toString() }


     static public func < (lhs: Car, rhs: Car) -> Bool
     {
          return lhs.compareTo(rhs) < 0
     }


     static public func == (lhs: Car, rhs: Car) -> Bool
     {
          return lhs.equals(rhs)
     }


    let model: String

    public init(_ model: String, _ year: Int){
    
        self.model = model
    super.init(year)
     }

    public init(_ model: String){
    
        self.model = model
    super.init()
     }

    public func equals(_ other: Any?) -> Bool
    {
        if nil == other || !(other is Car)
        {
            return false
        }

        let otherCar = other as! Car
        return self.model == otherCar.model && self.year == otherCar.year
    }

    public func compareTo(_ other: Car) -> Int
    {
        let comparison = self.year - other.year

        if comparison != 0
        {
            return comparison
        }

        return self.model.compareTo(other.model)
    }

    public func toString() -> String
    {
        return "My super car : \(self.model) on \(self.year)"
    }
}
